DashAI/back/models/hugging_face/distilbert_transformer.py

```diff
@@ -0,0 +1,176 @@
+    """
+    Pre-trained transformer DistilBERT allowing English text classification
+    """
+
+    @classmethod
+    def get_schema(cls):
+        with open("DashAI/back/models/parameters/models_schemas/DistilBERT.json") as f:
+            cls.SCHEMA = json.load(f)
+        return cls.SCHEMA
+
+    def __init__(self, model=None, **kwargs):
+        """
+        Initialize the transformer class by calling the pretrained model and its
+        tokenizer. Include an attribute analogous to sklearn's check_is_fitted to
+        see if it was fine-tuned.
+        """
+        self.model_name = "distilbert-base-uncased"
+        self.tokenizer = DistilBertTokenizer.from_pretrained(self.model_name)
+        self.model = (
+            model
+            if model is not None
+            else DistilBertForSequenceClassification.from_pretrained(self.model_name)
+        )
+        self.fitted = bool(model is not None)
+        self.training_args = kwargs
+
+    def get_tokenizer(self, input_column: str, output_column: str):
+        """Tokenize input and output
+
+        Parameters
+        ----------
+        input_column : str
+            name the input column to be tokenized
+        output_column : str
+            name the output column to be tokenized
+
+        Returns
+        -------
+        Function
+            Function for batch tokenization of the dataset
+        """
+
+        def tokenize(batch):
+            return {
+                "input_ids": self.tokenizer(
+                    batch[input_column],
+                    padding="max_length",
+                    truncation=True,
+                    max_length=512,
+                )["input_ids"],
+                "attention_mask": self.tokenizer(
+                    batch[input_column],
+                    padding="max_length",
+                    truncation=True,
+                    max_length=512,
+                )["attention_mask"],
+                "labels": batch[output_column],
+            }
+
+        return tokenize
+
+    def fit(self, dataset: DatasetDict):
+        """Fine-tuning the pre-trained model
+
+        Parameters
+        ----------
+        dataset : DatasetDict
+            Datasetdict with training data
+
+        """
+
+        train_dataset = dataset["train"]
+        input_column = train_dataset.inputs_columns[0]
+        output_column = train_dataset.outputs_columns[0]
+
+        tokenizer_func = self.get_tokenizer(input_column, output_column)
+        train_dataset = train_dataset.map(
+            tokenizer_func, batched=True, batch_size=len(train_dataset)
+        )
+        train_dataset.set_format(
+            "torch", columns=["input_ids", "attention_mask", "labels"]
+        )
+
+        # Arguments for fine-tuning
+        training_args = TrainingArguments(
+            output_dir="DashAI/back/user_models/temp_checkpoints_distilbert",
+            save_steps=1,
+            save_total_limit=1,
```

Comment:
I left it with those hyperparameters because for now, according to what we discussed with Felipe, we are only going to worry about the final state of the model and not the checkpoints, so this way we don't have several checkpoint folders lying around.